docs/tutorials/intro-to-rust.md

```diff
@@ -0,0 +1,418 @@
+
+For those who won't wish to dive into the deep end, consider this page a safe "wading pool" with no diving signs posted.
+
+## 2-Step Rust Installation
+
+### 1. Install Rustup
+
+`curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh`
+
+([Taken from official installation guide](https://www.rust-lang.org/tools/install))
+
+### 2. Add wasm target to your toolchain
+
+`rustup target add wasm32-unknown-unknown`
+
+<blockquote class="info">
+  <a href="https://doc.rust-lang.org/edition-guide/rust-2018/platform-and-target-support/webassembly-support.html" target="_blank">Why <code>unknown-unknown</code>?</a>
+</blockquote>
+
+## Resources
+
+The online book [from the official Rust site](https://doc.rust-lang.org/1.30.0/book/2018-edition/ch00-00-introduction.html) is a great resource to start with. It's recommended to read Chapter 1 before continuing, especially the `cargo` section.
+
+## Getting started
+
+Some smart contract examples from NEAR have a primary file that holds the code: `/src/lib.rs`. This is the conventional filename for a Rust library. Libraries will work great for compiling into WebAssembly and deployed the blockchain.
+
+The example shown here will use a `lib.rs` file with smart contract logic using a `struct`, the `struct`'s functions, and unit tests. This will all be in one file for this simple example. As developers build more complex smart contracts, it's a good idea to [organize code the Rust way](https://doc.rust-lang.org/1.30.0/book/2018-edition/ch12-03-improving-error-handling-and-modularity.html#separation-of-concerns-for-binary-projects).
+
+We'll break down the code in pieces below. If you wish to preview the complete code, please expand the bullets below.
+
+## Full example files
+
+<details>
+  <summary>Full `Cargo.toml` file</summary>
+
+```rust
+[package]
+name = "rust-counter-tutorial"
+version = "0.1.0"
+authors = ["Alice Bob <alice@example.com>"]
+edition = "2018"
+
+[lib]
+crate-type = ["cdylib", "rlib"]
+
+[dependencies]
+serde = { version = "1.0", features = ["derive"] }
+serde_json = "1.0.45"
+near-bindgen = "0.6.0"
+borsh = "0.6.1"
+wee_alloc = "0.4.5"
+
+[profile.release]
+codegen-units = 1
+# Tell `rustc` to optimize for small code size.
+opt-level = "z"
+lto = true
+debug = false
+panic = "abort"
+
+[workspace]
+members = []
+```
+
+</details>
+
+<details>
+  <summary>Full `src/lib.rs` file</summary>
+  
+```rust
+use borsh::{BorshDeserialize, BorshSerialize};
+use serde::{Deserialize, Serialize};
+use near_bindgen::{env, near_bindgen};
+
+#[global_allocator]
+static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
+
+// add the following attributes to prepare your code for serialization and invocation on the blockchain
+// More built-in Rust attributes here: https://doc.rust-lang.org/reference/attributes.html#built-in-attributes-index
+#[near_bindgen]
+#[derive(Default, BorshDeserialize, BorshSerialize)]
+#[derive(Serialize, Deserialize)]
```

Comment:
I wonder why this is necessary.

Reply:
I'm not sure and will lean on others for answers on this one.

Reply:
I was able to remove the serde import and derivces and everything works great, can't remove it from the `Cargo.toml` without an error. good eye.